tests/util/db_connection.py

```diff
@@ -1,19 +1,32 @@
 from __future__ import annotations
 
+import random
+import string
 import tempfile
 from contextlib import asynccontextmanager
 from pathlib import Path
-from typing import AsyncIterator
+from typing import AsyncIterator, Union
 
 from chia.util.db_wrapper import DBWrapper2
 
 
+def gen_in_memory_db_uri(prefix: str) -> str:
+    random_string = "".join(random.choice(string.ascii_letters) for _ in range(8))
+    return f"file:{prefix}{random_string}?mode=memory&cache=shared"
```

Comment:
I see you added `cache=shared` here, to make sure multiple DB connections will share the same underlying database. Is that the intention?